src/controllers/profileController.ts

```diff
@@ -0,0 +1,67 @@
+// controllers/userController.ts
+import { Request, Response } from 'express';
+import User from '../database/models/user';
+
+const getAllUserProfiles = async (req: Request, res: Response): Promise<void> => {
+  try {
+    // Find all user profiles
+    const users = await User.findAll();
+
+    // Return all user profiles
+    res.status(200).json(users);
+  } catch (error) {
+    // Handle errors
+    console.error('Error fetching user profiles:', error);
+    res.status(500).send('Internal Server Error');
+  }
+};
+
+const getUserProfile = async (req: Request, res: Response): Promise<void> => {
+  const userId: string = req.params.userId;
+
+  try {
+    // Find the user by ID
+    const user = await User.findByPk(userId);
+    if (!user) {
+      res.status(404).json({ message: 'User not found' });
+      return;
+    }
+
+    // Return user profile
+    res.status(200).json({
+      id: user.id,
+      firstName: user.firstName,
+      lastName: user.lastName,
+      email: user.email,
+    });
```

Comment:
Here, we need to retrieve a single user by their ID along with their roles. We require all the data pertaining to the user, considering it's a user profile. Therefore, you should fetch all the data for the user as well as any related data from other tables/models but for now it is just the model of Role only related.